errors.js

```diff
@@ -7,16 +7,32 @@ class ERR_INVALID_ARG_TYPE extends TypeError {
   }
 }
 
+class ERR_INVALID_OPTION_VALUE extends Error {
+  constructor(message) {
+    super(message);
+    this.code = 'ERR_INVALID_OPTION_VALUE';
+  }
+}
+
 class ERR_INVALID_SHORT_OPTION extends TypeError {
   constructor(longOption, shortOption) {
     super(`options.${longOption}.short must be a single character, got '${shortOption}'`);
     this.code = 'ERR_INVALID_SHORT_OPTION';
   }
 }
 
+class ERR_UNKNOWN_OPTION extends Error {
```

Comment:
Likewise, if we can avoid adding any new errors it simplifies merging into Node.js.

Reply:
see: https://github.com/pkgjs/parseargs/pull/97

Reply:
https://github.com/pkgjs/parseargs/pull/74#discussion_r846813662

Reply:
Is it possible to namespace our error codes so we can provide the most accurate messages to users?
e.g.
- `ERR_PARGS_*`
- `ERR_PARGS_UNKNOWN_OPTION`
    - Unknown long option --foo
    - Unknown short option -f
-  `ERR_PARGS_INVALID_OPTION_VALUE` -
    - Missing value for ${longOrShortOption} with type:'string'
    - Unexpected value '${optionValue}' for ${longOrShortOption} with type:'boolean'

Reply:
There appear to be error namespaces in node core:
- [ERR_IPC_*](https://github.com/nodejs/node/blob/ec5a359ffd38b6f56a456a0b6a482b5167fbfce1/lib/internal/errors.js#L1356-L1359)
- [ERR_MANIFEST_*](https://github.com/nodejs/node/blob/ec5a359ffd38b6f56a456a0b6a482b5167fbfce1/lib/internal/errors.js#L1360-L1391)
- [ERR_SOCKET_*](https://github.com/nodejs/node/blob/ec5a359ffd38b6f56a456a0b6a482b5167fbfce1/lib/internal/errors.js#L1512-L1529)
- [ERR_STREAM_*](https://github.com/nodejs/node/blob/ec5a359ffd38b6f56a456a0b6a482b5167fbfce1/lib/internal/errors.js#L1533-L1544)
- etc.

@bcoe @shadowspawn @shadowspawn What are your thoughts on using a namespace for `parseArgs` errors. For example `ERR_PARGS_*` or `ERR_PARSEARGS_*`

Reply:
Updated unique `parseArgs` error classes to `ERR_PARSE_ARGS_*`: https://github.com/pkgjs/parseargs/pull/74/commits/694c75d849e60f470d7c5aaa0fe929f9e64a7249